.github/workflows/ci.yml

```diff
@@ -1,72 +1,120 @@
-        run: |
-          node --version
-          npm --version
-
-      - name: Install dependencies
-        run: npm install
-
-      - name: Cache examples
-        uses: actions/cache@v4
+      - name: Set up the repo
+        uses: tree-sitter/parser-setup-action@v1.2
+      - name: Set up examples
+        run: |-
+          git clone https://github.com/0install/0install examples/0install --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/BinaryAnalysisPlatform/bap examples/bap --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/dbuenzli/cmdliner examples/cmdliner --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/facebook/flow examples/flow --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/facebook/pyre-check examples/pyre-check --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/garrigue/lablgtk examples/lablgtk --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/janestreet/base examples/base --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/mirage/ocaml-cohttp examples/ocaml-cohttp --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml/dune examples/dune --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml/merlin examples/merlin --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml/ocaml examples/ocaml --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml/ocaml-lsp examples/ocaml-lsp --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml/opam examples/opam --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml-multicore/eio examples/eio --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml-ppx/ocamlformat examples/ocamlformat --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocaml-ppx/ppxlib examples/ppxlib --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocsigen/js_of_ocaml examples/js_of_ocaml --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/ocsigen/lwt examples/lwt --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/owlbarn/owl examples/owl --single-branch --depth=1 --filter=blob:none
+      - name: Run tests
+        uses: tree-sitter/parser-test-action@v1.2
         with:
-          path: examples
-          key: ${{ hashFiles('script/parse-examples') }}
-
-      - name: Test corpus & parse examples
-        run: npm test
+          test-library: ${{runner.os == 'Linux'}}
+          corpus-files: |
+            examples/**/*.ml
+            examples/**/*.mli
+          invalid-files: |
```

Comment:
The `invalid-files` list is for files that actually contain syntax errors.
Files that are not yet parsed correctly should be in `corpus-files` prefixed with `!`.

Reply:
There are no (known) valid files that can't be parsed by `tree-sitter-ocaml`. The files listed here actually do contain invalid syntax. Most of them are either test files for development tools that contain invalid syntax intentionally, or are using a custom preprocessor.

But if it's really the intention to handle these cases differently, maybe it's best to document that. Why should files that can't be parsed yet be silently ignored, and files with invalid syntax give a warning? Shouldn't it be the other way around?

Reply:
Files that can't be parsed yet may be ignored to avoid failing CI.
Files with invalid syntax warn only if they are parsed as valid.